### DISCUSSION TITLE
Connections cache MAXAGE

### DISCUSSION THREAD

**Discussion by myudaZS** [curl/curl#12929] [Q&A] [answered]
Hi,
Questions about CURLOPT_MAXAGE_CONN for a single connection:
1. How CURLOPT_MAXAGE_CONN idle time is counted? Is it the time in seconds from the last curl_easy_perform call?
2. In order to close a connection that it's age exceeded maxage, should the user invoke  curl_easy_perform / curl_easy_cleanup, or does curl has an internal timer to close old connections without the user intervention. 
3. In case of  connectivity issues to a specific host at the second call, will the connection still be reused at the third call, assuming its age < maxage at the time of the third call?

**Comment by bagder** [comment, marked as answer]
> How CURLOPT_MAXAGE_CONN idle time is counted? Is it the time in seconds from the last curl_easy_perform call?

It is the time since it was most recently used by a transfer.

> In order to close a connection that it's age exceeded maxage, should the user invoke curl_easy_perform / curl_easy_cleanup, or does curl has an internal timer to close old connections without the user intervention.

curl can only close connections when it runs. Most commonly, the too old connections are closed when you start a new transfer.

> In case of connectivity issues to a specific host at the second call, will the connection still be reused at the third call, assuming its age < maxage at the time of the third call?

if there exists a live connection to reuse "at the second call", how can there also be "connectivity issues" ?
